core/state_transition.go

```diff
@@ -104,10 +106,10 @@ func (result *ExecutionResult) Revert() []byte {
 }
 
 // IntrinsicGas computes the 'intrinsic gas' for a message with the given data.
-func IntrinsicGas(data []byte, contractCreation, isHomestead bool, isEIP2028 bool) (uint64, error) {
+func IntrinsicGas(data []byte, accessList *types.AccessList, isContractCreation bool, isHomestead, isEIP2028, isEIP2718 bool) (uint64, error) {
```

Comment:
I'm a bit torn here. Does it really make sense to have `isEIP2718`? If this method instead takes a (potentially nil) accessList, then we simply charge for it. 
As it is, if we ask about the intrinsic gas for an accesslist-type tx but say that it's not 2718, then this method will return the wrong intrinsic gas. 
If we skip `isEIP2718`, then at least the method cannot return values that doesn't make sense

Reply:
That's a fair point. It's also poor wording, `is2718` is more akin to `isHomestead`. But I think you're right, it's okay to simply use the `accessList` value.